e2store/src/bin/era2-stats.rs

```diff
@@ -61,6 +78,24 @@ fn main() -> anyhow::Result<()> {
             };
             stats.storage_entry_count += 1;
             stats.storage_item_count += storage.len();
+
+            if index + 1 < account.storage_count {
+                assert_eq!(
+                    storage.len(),
+                    10_000_000,
+                    "Non-final storage entry doesn't have 10M items",
```

Comment:
A bit more info in the error message would be sweet. Probably something like: the storage index, the actual storage length and which account it was in.